File: src/真题/noip/noip2006/普及组/明明的随机数/Main.cpp
// https://www.luogu.com.cn/problem/P1059
#include <bits/stdc++.h>
using namespace std;
int main() {
  int n;
  cin >> n;
  set<int> s;
  int arr[n];
  int c = 0;
  for (int i = 0; i < n; i++) {
    int tmp;
    cin >> tmp;
    if (s.insert(tmp).second == true) {
      arr[c++] = tmp;
    }
  }
  sort(arr, arr + c);
  cout << s.size() << endl;
  for (int i = 0; i < c; i++) {
    cout << arr[i] << ' ';
  }
  return 0;
}

File: src/真题/noip/noip2006/普及组/明明的随机数/Main.java
// https://www.luogu.com.cn/problem/P1059
package 真题.noip.noip2006.普及组.明明的随机数;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    HashSet<Integer> set = new HashSet<>();
    int[] arr = new int[n];
    int tmp;
    for (int i = 0; i < n; i++) {
      tmp = sc.nextInt();
      if (set.add(tmp)) {
        arr[i] = tmp;
      }
    }
    System.out.println(set.size());
    Arrays.sort(arr);
    for (int i = 0; i < n; i++) {
      if (arr[i] == 0) {
        continue;
      }
      System.out.print(arr[i] + " ");
    }
    sc.close();
  }
}
